src/games/strategy/engine/data/GameParser.java

```diff
@@ -984,30 +998,40 @@ private void parseEditableProperty(final Element property, final String name, fi
     final Element child = (Element) children.get(0);
     final String childName = child.getNodeName();
     IEditableProperty editableProperty;
-    if (childName.equals("boolean")) {
-      editableProperty = new BooleanProperty(name, null, Boolean.valueOf(defaultValue).booleanValue());
-    } else if (childName.equals("file")) {
-      editableProperty = new FileProperty(name, null, defaultValue);
-    } else if (childName.equals("list") || childName.equals("combo")) {
-      final StringTokenizer tokenizer = new StringTokenizer(child.getAttribute("values"), ",");
-      final Collection<String> values = new ArrayList<>();
-      while (tokenizer.hasMoreElements()) {
-        values.add(tokenizer.nextToken());
+    switch (childName) {
```

Comment:
Not sure this is actually easier to read/maintain converting to a switch. Once you have logic in checks where multiple cases map to the same logic, its probably easier to read as if/else.